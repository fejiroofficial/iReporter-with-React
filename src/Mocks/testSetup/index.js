/* eslint-disable import/no-extraneous-dependencies */
import '@babel/polyfill';
import React from 'react';
import { Provider } from 'react-redux';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import configMockStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';


export const mockState = {
  user: {
    token: '',
    isLoggedIn: false,
    loginStatus: false,
    message: '',
    isLoading: false,
    profile: {
      firstname: 'John',
      username: 'johnny',
      image: 'image.jpg',
      errorData: [],
    },
    resetPassword: {
      loading: false,
      message: '',
    }
  },
  bookmark: {
    loading: false,
    articles: [],
    message: '',
  },
  article: {
    newComments: {
      body: 'Hello world',
      id: 1,
      authorName: 'John',
      authorImage: 'image.jpg',
      date: 'July 4th 2018',
    },
    postingComment: false,
    commentMessage: 'Done',
    gettingComments: false,
    oldComments: [],
    hasMoreComments: false,
    commentPage: {
      current: 1,
      currentCount: 1,
      totalCount: 1,
    },
  },
  createArticle: {
    loading: false,
    message: false,
  },
  articlesReducer: {
    articles: []
  },
  articlesCategoryReducer: []
};


export const createMockStore = () => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  return mockStore;
};

const setup = (component, initailState = mockState) => {
  const mockStore = configMockStore([thunk]);
  configure({ adapter: new Adapter() });
  const connectedWrapper = mount(
    <Provider store={mockStore(initailState)}>
      <BrowserRouter>
        {component}
      </BrowserRouter>
    </Provider>
  );
  return connectedWrapper;
};

export default setup;
