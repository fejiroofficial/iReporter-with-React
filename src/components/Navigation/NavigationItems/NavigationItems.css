.LargeScreenNav {
    display: none;
}

.LargeScreenNav a{
    padding: 8px;
    padding-top: 15px;
    padding-left: 22px;
    padding-right: 22px;
    text-decoration: none;
    font-size: 14px;
    font-weight: bold;
    color: ghostwhite;
}

@media (min-width: 1024px) {
    .LargeScreenNav  {
        display: flex;
        height: 47px;
        padding-left: 69px;
        background-color: lightseagreen;
    }
    .Seperator {
        height: inherit;
        width: 1px;
        background-color: ghostwhite;
    }
}

@media (min-width: 1359px) {
    .LargeScreenNav {
        display: flex;
        height: 47px;
        padding-left: 187px;
        background-color: lightseagreen;
    }
}
