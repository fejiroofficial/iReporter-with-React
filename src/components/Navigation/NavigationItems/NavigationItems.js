import React from 'react';
import { NavLink } from 'react-router-dom';
import Classes from './NavigationItems.css';

const NavigationItems = (props) => (
  <div id="myNav" className={Classes.LargeScreenNav}>
    <div className={Classes.Seperator}></div>
    <NavLink to="/about" >ABOUT US</NavLink>
    <div className={Classes.Seperator}></div>
    <NavLink to="/corruption" >LEARN ABOUT CORRUPTION</NavLink>
    <div className={Classes.Seperator}></div>
    <NavLink to="/ireport/red-flags">REDFLAGS</NavLink>
    <div className={Classes.Seperator}></div>
    <NavLink to="/ireport/interventions">INTERVENTIONS</NavLink>
    <div className={Classes.Seperator}></div>
    <NavLink to="/ireport" >IREPORT</NavLink>
    <div className={Classes.Seperator}></div>
  </div>
);

export default NavigationItems;
