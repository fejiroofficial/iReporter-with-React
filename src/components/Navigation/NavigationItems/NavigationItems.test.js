import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { createMockStore } from '../../../Mocks/testSetup';
import NavigationItems from './NavigationItems';

createMockStore();

describe('<NavigationItems>', () => {
  it('should render without crashing', () => {
    const container = (
      <NavigationItems>
        <p>Hello</p>
      </NavigationItems>
    );
    const wrapper = shallow(container);
    expect(wrapper).toMatchSnapshot();
  });
});
