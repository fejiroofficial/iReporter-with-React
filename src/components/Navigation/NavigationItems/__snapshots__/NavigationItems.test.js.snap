// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems> should render without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
